### ISSUE TITLE
Utilize use_extern_macros

### ISSUE THREAD

**Issue by breeswish** [pingcap/tikv#3356]
## Feature Request

Currently our macros are exported globally. We can make it modular by using the feature "use_extern_macros" so that unnecessary macros are not visible in the scope.

However there some commonly used macros, i.e. info!, warn!. We may create something like "common_macros.rs" and `use ::common_macros::*` so that it can be easy to use.

**Comment by brson** [comment]
Completing https://github.com/tikv/tikv/issues/4196 will cover this for macros defined in dependencies.

For our own internal macros is it your intent @breeswish that we move them into their own crate? For internal macros I think the only option remains putting a `#[macro_use]` on their module, which exports them globally. At least I think that's the behavior of `#[macro_use]` on modules.

**Comment by breeswish** [comment]
@brson Not really. After a long time now I think that for those really global macros looks like `#[macro_use]` is still a good choice (for example, info!, warn!, etc).
